src/app/home/event-user-list/event-user-list.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit, Input, ViewChild, Output, EventEmitter } from '@angular/core';
+import { ClrModal } from '@clr/angular';
+import { Progress } from 'src/app/data/progress';
+import { User } from 'src/app/data/user';
+
+interface userListUser extends User {
+  numberOfSessions?: number;
+  activeScenarioName?: string;
+}
+
+@Component({
+  selector: 'event-user-list',
+  templateUrl: './event-user-list.component.html'  
+})
+export class EventUserListComponent implements OnInit {
+
+  @ViewChild("userListModal") userListModal: ClrModal;
+
+  @Input()
+  public users: User[];
+
+  @Input()
+  public progress: Progress[]; 
+
+  @Input()
+  public filterName: Function;
+
+  @Output() nameClickedEvent = new EventEmitter<string>();
+  
+  public userListUsers: userListUser[] = [];
+
+  ngOnInit(): void {}
+
+  public listFilterName(username) {
```

Comment:
Frankly, I do not know what to expect from this method by just reading the name. Since it actually does two quite different things, the easy way would be to name it according to its usage instead, i.e. `onUserSelect` or something similar. But that's just my 2 cents.